app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -39,17 +42,37 @@ class AllFilesFragment : Fragment() {
     @RequiresApi(Build.VERSION_CODES.R)
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        val items = mutableListOf<ListModel>()
-        val path = Environment.getRootDirectory().absolutePath
-        File(path).walk().forEach {
-            items.add(ListModel(it.name))
-        }
 
-        adapter = ListAdapter()
         val layoutManager = LinearLayoutManager(context)
+        adapter = ListAdapter(requireContext(), fetchList(), this)
         binding.recyclerView.layoutManager = layoutManager
-        binding.recyclerView.itemAnimator = DefaultItemAnimator()
-        adapter.replaceItems(items)
         binding.recyclerView.adapter = adapter
+        binding.recyclerView.addItemDecoration(
+            DividerItemDecoration(
+                context,
+                RecyclerView.VERTICAL
+            )
+        )
+    }
+
+    @RequiresApi(Build.VERSION_CODES.R)
```

Comment:
Why is this annotation required? Please make it backwards compatble, it should work on Android version 24 as well.

Reply:
It is required for Environment.getStorageDirectory() method

Reply:
That means that it will work only on android R sdk version 30. Could you please investigate how to implement it to work on Android version 24 as well?